crates/sui-core/src/execution_engine.rs

```diff
@@ -319,13 +323,24 @@ fn transfer_sui<S>(
         // This is necessary for the temporary store to know this new object is not unwrapped.
         let newly_generated_ids = tx_ctx.recreate_all_ids();
         temporary_store.set_create_object_ids(newly_generated_ids);
+        Some(amount)
     } else {
         // If amount is not specified, we simply transfer the entire coin object.
         // We don't want to increment the version number yet because latter gas charge will do it.
         object.transfer_without_version_change(recipient);
-    }
+        Coin::extract_balance_if_coin(&object)?
+    };
 
-    // TODO: Emit a new event type for this.
+    temporary_store.log_event(Event::TransferObject {
+        package_id: ObjectID::from(SUI_FRAMEWORK_ADDRESS),
+        transaction_module: Identifier::from(ident_str!("native")),
+        sender: tx_ctx.sender(),
+        recipient: Owner::AddressOwner(recipient),
+        object_id: object.id(),
+        version: object.version(),
+        type_: TransferType::Coin, // Should this be a separate type, like SuiCoin?
```

Comment:
Can you resolve this comment? Not sure if I understand this question correctly, but this event is emitted for non-sui coins(e.g., coin::Coin<USDC>) right?

Reply:
Oh. Maybe it's a good question for you.  I can just remove the comment, the question was whether we need a different transfer type for Sui coins vs non-Sui coins.    That code is in `transfer_sui()` which is I believe only for Sui coins?
So the question is do we need a separate transfer type for Sui coins vs non Sui 'coin::Coin` types.  For now I just leave it the same.